### PR TITLE
query and update animals table

### PR DESCRIPTION
**Pull request by christianbiring1**
In this project, I had to use transactions to update and delete records. I have also used my knowledge of aggregate functions and GROUP BY to answer analytical questions.

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -8,3 +8,5 @@ CREATE TABLE animals (
     neutered boolean,
     weight_kg decimal
 );
+
+ALTER TABLE vet_clinicschema.animals ADD COLUMN species varchar;
```
Comment (DeliceLydia) [inline schema.sql]: - [ ] Please add varchar limit may be(255)

File: queries.sql [hunk 2]
```diff
@@ -8,3 +8,33 @@ SELECT name, escape_attempts from animals WHERE weight_kg > 10.5;
 SELECT * from animals WHERE neutered = TRUE;
 SELECT * from animals WHERE name != 'Gabumon';
 SELECT * from animals WHERE 10.4 <= weight_kg AND weight_kg >= 17.3;
+
+BEGIN;
+ALTER TABLE animals RENAME COLUMN species TO unspecified;
+SELECT * FROM animals;
```
Comment (DeliceLydia) [inline queries.sql:14]: - [ ] After making species `unspecified` you are supposed to verify that change was made
```sql
SELECT species from animals; -- verify that change was made
   ROLLBACK;

   SELECT species from animals; -- verify that change was undone
```

File: schema.sql (hunk 1 above)
Comment (DeliceLydia) [reply schema.sql]: - [ ] Please add these queries:
- `UPDATE` the animal's table by setting the species column to Digimon for all animals that have a name ending in mon.
- `UPDATE` the animals table by setting the species column to pokemon for all animals that don't have species already set.
```sql
BEGIN;
   UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; 
   UPDATE animals SET species = 'pokemon' WHERE species IS NULL; 

   SELECT species from animals; -- verify that change was made
   COMMIT;
   SELECT species from animals; -- verify that change persists after commit
```

File: queries.sql [hunk 3]
```diff
@@ -8,3 +8,33 @@ SELECT name, escape_attempts from animals WHERE weight_kg > 10.5;
...
+
+-- What is the minimum and maximum weight of each type of animal?
+SELECT MIN (weight_kg) FROM animals WHERE species = 'pokemon';
+SELECT MAX (weight_kg) FROM animals WHERE species = 'pokemon';
+SELECT MIN (weight_kg) FROM animals WHERE species = 'digimon';
+SELECT MAX (weight_kg) FROM animals WHERE species = 'digimon';
+
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT AVG (escape_attempts) FROM animals WHERE species = 'digimon' AND date_of_birth between '1990-01-01' AND '2000-12-31';
+SELECT AVG (escape_attempts) FROM animals WHERE species = 'pokemon' AND date_of_birth between '1990-01-01' AND '2000-12-31';
```
Comment (DeliceLydia) [inline queries.sql:66]: - [ ] Please add a query Inside a transaction that delete all records in the animal's table, then roll back the transaction
```sql
BEGIN;

   DELETE FROM animals WHERE ;
   SELECT COUNT(*) FROM ANIMALS;

   ROLLBACK;
   SELECT COUNT(*) FROM ANIMALS;
```

Comment (DeliceLydia) [reply queries.sql:66]: - [ ] Add a query that
- `DELETE` all animals born after Jan 1st, 2022
- Create a savepoint for the transaction
- `UPDATE` all animals' weight to be their weight multiplied by -1.
- Rollback to the savepoint
- `UPDATE` all animals' weights that are negative to be their weight multiplied by -1.
- Commit transaction
```sql
BEGIN;

    DELETE FROM animals WHERE  date_of_birth >= '2022/01/01' ;
    SAVEPOINT SP1;

    UPDATE animals SET weight_kg = weight_kg * -1; 
    ROLLBACK TO SP1;
    UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0; 

    COMMIT;
```

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @christianbiring1, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._
- [ ] Remember to add a screenshots of their queries results to the pull request description


### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: schema.sql (hunk 1 above)
Comment (christianbiring1) [reply schema.sql]: > * [ ]  Please add varchar limit may be(255)

:+1: :rocket:

Comment (christianbiring1) [reply schema.sql]: > * [ ]  Please add these queries:
> * `UPDATE` the animal's table by setting the species column to Digimon for all animals that have a name ending in mon.
> * `UPDATE` the animals table by setting the species column to pokemon for all animals that don't have species already set.
> 
> ```sql
> BEGIN;
>    UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; 
>    UPDATE animals SET species = 'pokemon' WHERE species IS NULL; 
> 
>    SELECT species from animals; -- verify that change was made
>    COMMIT;
>    SELECT species from animals; -- verify that change persists after commit
> ```

I have actually forgotten to put all those stuff in the sql file in the repos. :smiley:

File: queries.sql (hunk 2 above)
Comment (christianbiring1) [reply queries.sql:14]: Yeah yeah! I have forgotten to put all those lines in the repos.

File: queries.sql (hunk 3 above)
Comment (christianbiring1) [reply queries.sql:66]: I have forgotten to put those lines in my repos. I was in a rush a little bit...

Comment (christianbiring1) [reply queries.sql:66]: > * [ ]  Please add a query Inside a transaction that delete all records in the animal's table, then roll back the transaction
> 
> ```sql
> BEGIN;
> 
>    DELETE FROM animals WHERE ;
>    SELECT COUNT(*) FROM ANIMALS;
> 
>    ROLLBACK;
>    SELECT COUNT(*) FROM ANIMALS;
> ```
For checking those data we weren't supposed to do so in a transition as long as they don't affect the existing ones. It was only for calculation purposes. so no need to grab the queries within a transition. @DeliceLydia.

**Review by DeliceLydia — APPROVED** [review]
## Hi @christianbiring1, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**